test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -7382,6 +7382,33 @@ BOOST_AUTO_TEST_CASE(inline_assembly_function_access)
 	BOOST_CHECK(callContractFunction("x()") == encodeArgs(u256(10)));
 }
 
+BOOST_AUTO_TEST_CASE(inline_assembly_labels_with_stack_info)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f(uint y) {
+				assembly {
+					y 7
+					jump(fun)
+					exit[-1]:
+					y add
+					0 mstore
+					return(0, 0x20)
+					fun:
+					{
+						entry[a, b]:
+						a := div(a, b)
```

Comment:
What is the rationale behind the choice that `b` remains in the scope after it is used.  Because of this choice, there needs to be an implicit `DUP`, which need to be taken into mental account of the stack heights.  It is much simpler if one variable can be only used once, and then the programmers can forget about the identifier.  That also removes the need for the next `pop`.